<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Play rosbag record -->
    <arg name="bagfile" default="lidar_cam_calib4.bag"/>
    <node name="rosbag_player" pkg="rosbag" type="play"
          args="--clock --loop $(find Camera_Lidar)/DATA/$(arg bagfile)"/>

    <!-- Start rviz -->
    <!--<arg name="rvizfile" default="Test.rviz"/>
    <node type="rviz" name="rviz" pkg="rviz"
          args="-d $(find Camera_Lidar)/DATA/$(arg rvizfile)"/> -->

    <!-- Camera & Lidar subscriber node -->
    <arg name="image_topic_name" default="/pylon_camera_node/image_raw"/>
    <arg name="info_topic_name" default="/pylon_camera_node/cam_info"/>
    <arg name="lidar_topic_name" default="/velodyne_points"/>

    <node name="cam_laser" pkg="Camera_Lidar" type="main.py" output="screen">
        <param name="image_topic_name" type="str" value="$(arg image_topic_name)"/>
        <param name="info_topic_name" type="str" value="$(arg info_topic_name)"/>
        <param name="lidar_topic_name" type="str" value="$(arg lidar_topic_name)"/>
    </node>

</launch>